src/postgres/commands/PostgresAccountWizard/PostgresServerNameStep.ts

```diff
@@ -43,11 +42,17 @@ async function validatePostgresServerName(name: string, client: PostgreSQLManage
     }
 
     const availabilityRequest: NameAvailabilityRequest = { name: name, type: "Microsoft.DBforPostgreSQL" };
+    const characterMatch = name.match(/^[a-z\d-]+$/);
     const availability: NameAvailability = (await client.checkNameAvailability.execute(availabilityRequest));
 
     if (!availability.nameAvailable) {
-
-        return availability.message;
+        if (availability.reason === 'AlreadyExists') {
+            return localize('serverNameAvailabilityCheck', 'Server name "{0}" is not available.', name);
+        } else if (!characterMatch) {
```

Comment:
Seems like Will and I both prefer to have the RegExp in the condition itself instead of a separate variable. To me, it makes it more read-able to have the RegExp and the message right next to each other. I also just prefer fewer characters when possible haha 🤷‍♂️ Any particular reason you prefer the variable?

Reply:
Lol 😅 you mentioned in earlier suggestion to place the check before await to reduce delay https://github.com/microsoft/vscode-cosmosdb/pull/1447#discussion_r419774990

Reply:
Oh yes that's correct, but actually your existing code doesn't reduce delay since you await the call to check name availability before you return the character check message. You need to do something like this:

```
if (name.length < min || name.length > max) {
    return localize('serverNameLengthCheck', 'The name must be between {0} and {1} characters.', min, max);
} else if (!name.match(/^[a-z\d-]+$/)) {
    return localize('serverNameCharacterCheck', 'Server name must only contain lowercase letters, numbers, and hyphens.');
} else if (name.startsWith('-') || name.endsWith('-')) {
    return localize('serverNamePrefixSuffixCheck', 'Server name must not start or end in a hyphen.');
} else {
    const availabilityRequest: NameAvailabilityRequest = { name: name, type: "Microsoft.DBforPostgreSQL" };
    const availability: NameAvailability = (await client.checkNameAvailability.execute(availabilityRequest));
    if (!availability.nameAvailable) {
        if (availability.reason === 'AlreadyExists') {
            return localize('serverNameAvailabilityCheck', 'Server name "{0}" is not available.', name);
        } else {
            return availability.message;
        }
    }
}
```

It can't even reach the async code if it has invalid characters.

Reply:
You're right, for some reason I was thinking I needed NameAvailability for the regex check